gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -119,12 +738,1491 @@ gum_stalker_add_call_probe (GumStalker * self,
+      if (insn->ci->id == ARM64_INS_CBZ)
+        gum_arm64_writer_put_cbnz_reg_label (cw, op->reg,  is_false);
+      else if (insn->ci->id == ARM64_INS_CBNZ)
+        gum_arm64_writer_put_cbz_reg_label (cw, op->reg,  is_false);
+      else if (insn->ci->id == ARM64_INS_TBZ)
+        gum_arm64_writer_put_tbnz_reg_imm_label (cw, op->reg, op2->imm,
+            is_false);
+      else if (insn->ci->id == ARM64_INS_TBNZ)
+        gum_arm64_writer_put_tbz_reg_imm_label (cw, op->reg, op2->imm,
+            is_false);
+      else if(insn->ci->id == ARM64_INS_B)
+      {
+        g_assert (cc != ARM64_CC_INVALID);
+        g_assert (cc > ARM64_CC_INVALID);
+        g_assert (cc <= ARM64_CC_NV);
+        arm64_cc not_cc = cc + 2*(cc%2) -1;
+#if DEBUG_LEVEL == 10
+        g_print ("cc (%d) and not_cc (%d)",cc, not_cc);
+#endif
+        gum_arm64_writer_put_b_cond_label (cw, not_cc, is_false);
+      }
+      else g_assert_not_reached ();
+    }
+
+    gum_exec_block_write_jmp_transfer_code (block, &target, gc);
+
+    if (is_conditional)
+    {
+      GumBranchTarget cond_target = { 0, };
+
+      cond_target.is_indirect = FALSE;
+      cond_target.absolute_address = insn->end;
+
+      gum_arm64_writer_put_label (cw, is_false);
+
+      gum_exec_block_write_jmp_transfer_code (block, &cond_target, gc);
+    }
+  }
+  else
+  {
+    g_assert ("" == "not implemented");
+  }
+
+  return GUM_REQUIRE_NOTHING;
+}
+
+static GumVirtualizationRequirements
+gum_exec_block_virtualize_ret_insn (GumExecBlock * block,
+                                    GumGeneratorContext * gc)
+{
+  if ((block->ctx->sink_mask & GUM_RET) != 0)
+    gum_exec_block_write_ret_event_code (block, gc, GUM_CODE_INTERRUPTIBLE);
+
+  gum_arm64_relocator_skip_one (gc->relocator);
+  gum_exec_block_write_ret_transfer_code (block, gc);
+
+  return GUM_REQUIRE_NOTHING;
+}
+
+static GumVirtualizationRequirements
+gum_exec_block_virtualize_sysenter_insn (GumExecBlock * block,
+                                         GumGeneratorContext * gc)
+{
+  (void) block;
+  (void) gc;
+
+  return GUM_REQUIRE_RELOCATION;
+}
+
+static void
+gum_exec_block_write_call_invoke_code (GumExecBlock * block,
+                                       const GumBranchTarget * target,
+                                       GumGeneratorContext * gc)
+{
+#if DEBUG_LEVEL == 10
+  g_print ("gum_exec_block_write_call_invoke_code - enter\n");
+#endif
+
+  gboolean can_backpatch;
+  GumArm64Writer * cw = gc->code_writer;
+  gpointer call_code_start;
+  GumPrologType opened_prolog;
+
+  call_code_start = cw->code;
+  opened_prolog = gc->opened_prolog;
+
+  gum_exec_block_open_prolog (block, GUM_PROLOG_MINIMAL, gc);
+
+#if DEBUG_LEVEL == 10
+  g_print (
+      "\t> gum_exec_ctx_replace_current_block_with\n\t\t(block->ctx: %p,\n\t\ttarget->absolute_address: %p|%d)\n",
+      block->ctx, target->absolute_address, target->base);
+#endif
+
+  /* generate code for the target */
+  /* get the target */
```

Comment:
Use a multi-line comment here (see earlier comment for details about its style). Also the comments here are confusing me.